src/include/execution/vm/bytecode_handlers.h

```diff
@@ -1321,215 +1323,202 @@ VM_OP void OpIndexIteratorPerformInit(terrier::execution::sql::IndexIterator *it
-
-  // Set
-  out->is_null_ = null;
-  out->val_ = null ? 0 : *ptr;
+VM_OP_HOT void OpPRSetVarlenNull(terrier::execution::sql::ProjectedRowWrapper *pr, uint16_t col_idx,
+                                 terrier::execution::sql::StringVal *val) {
+  if (val->len_ <= terrier::storage::VarlenEntry::InlineThreshold()) {
+    terrier::storage::VarlenEntry varlen(terrier::storage::VarlenEntry::CreateInline(
+        reinterpret_cast<const terrier::byte *>(val->Content()), val->len_));
+    pr->Set<terrier::storage::VarlenEntry, true>(col_idx, varlen, val->is_null_);
+  } else {
+    terrier::storage::VarlenEntry varlen(terrier::storage::VarlenEntry::Create(
+        reinterpret_cast<const terrier::byte *>(val->Content()), val->len_, false));
+    pr->Set<terrier::storage::VarlenEntry, true>(col_idx, varlen, val->is_null_);
+  }
 }
 
-VM_OP_HOT void OpIndexIteratorGetSmallIntNull(terrier::execution::sql::Integer *out,
-                                              terrier::execution::sql::IndexIterator *iter, uint16_t col_idx) {
-  // Read
-  bool null = false;
-  auto *ptr = iter->Get<int16_t, true>(col_idx, &null);
+VM_OP_HOT void OpPRSetDate(terrier::execution::sql::ProjectedRowWrapper *pr, uint16_t col_idx,
+                           terrier::execution::sql::Date *val) {
+  pr->Set<uint32_t, false>(col_idx, val->int_val_, val->is_null_);
+}
 
-  // Set
-  out->is_null_ = null;
-  out->val_ = null ? 0 : *ptr;
+VM_OP_HOT void OpPRSetDateNull(terrier::execution::sql::ProjectedRowWrapper *pr, uint16_t col_idx,
+                               terrier::execution::sql::Date *val) {
+  pr->Set<uint32_t, true>(col_idx, val->int_val_, val->is_null_);
 }
 
-VM_OP_HOT void OpIndexIteratorGetIntegerNull(terrier::execution::sql::Integer *out,
-                                             terrier::execution::sql::IndexIterator *iter, uint16_t col_idx) {
+VM_OP_HOT void OpPRGetDate(terrier::execution::sql::Date *out, terrier::execution::sql::ProjectedRowWrapper *pr,
+                           uint16_t col_idx) {
   // Read
-  bool null = false;
-  auto *ptr = iter->Get<int32_t, true>(col_idx, &null);
-
+  auto *ptr = pr->Get<uint32_t, false>(col_idx, nullptr);
+  TERRIER_ASSERT(ptr != nullptr, "Null pointer when trying to read date");
   // Set
-  out->is_null_ = null;
-  out->val_ = null ? 0 : *ptr;
+  out->is_null_ = false;
+  out->int_val_ = *ptr;
 }
 
-VM_OP_HOT void OpIndexIteratorGetBigIntNull(terrier::execution::sql::Integer *out,
-                                            terrier::execution::sql::IndexIterator *iter, uint16_t col_idx) {
+VM_OP_HOT void OpPRGetDateNull(terrier::execution::sql::Date *out, terrier::execution::sql::ProjectedRowWrapper *pr,
+                               uint16_t col_idx) {
   // Read
   bool null = false;
-  auto *ptr = iter->Get<int64_t, true>(col_idx, &null);
+  auto *ptr = pr->Get<uint32_t, true>(col_idx, &null);
+  TERRIER_ASSERT(ptr != nullptr, "Null pointer when trying to read date");
```

Comment:
You can't do this assert if the type is nullable (assuming that's what the function signature means). Null attributes return nullptr when reading from the storage layer.